src/charm.py

```diff
@@ -83,8 +88,20 @@ def __init__(self, *args):
         self.redis = RedisRequires(self, self._stored)
         self.framework.observe(self.on.redis_relation_updated, self._config_changed)
 
-    def _make_ingress_config(self):
-        """Return a dict of our ingress config."""
+        self._metrics_endpoint = MetricsEndpointProvider(
+            self, jobs=[{"static_configs": [{"targets": ["*:9394"]}]}]
+        )
+        self._logging = LogProxyConsumer(
+            self, relation_name="logging", log_files=LOG_PATHS, container_name="discourse"
+        )
+        self._grafana_dashboards = GrafanaDashboardProvider(self)
+
+    def _make_ingress_config(self) -> Dict:
```

Comment:
It is possible to define the key and value type, although it is not always useful, using `Dict[str, str]`. If it gets too complex, it is also possible to use TypedDict: https://peps.python.org/pep-0589/

Reply:
I'll try to be a little more specific in these cases.